### ISSUE TITLE
CA1507 has faulty heuristic

### ISSUE THREAD

**Issue by kentcb** [dotnet/roslyn#23601]
**Version Used**: 

2.6.0-beta1 running in VS 15.5.

**Steps to Reproduce**:

This code:
```C#
public JourneyMode? JourneyMode
{
    get => this.journeyMode;
    set => this.RaiseAndSetIfChanged(ref this.journeyMode, value);
}
```

Triggers this warning:
```
Use nameof in place of string literal 'JourneyMode'
```

Applying the faulty fix results in this (invalid) code:

```C#
public JourneyMode? JourneyMode
{
    get => this.journeyMode;
    set => nameof(@this);
}
```

It seems to stem from the fact that `RaiseAndSetIfChanged` leverages the `[CallerMemberName]` attribute to gain access to the caller's name.

**Expected Behavior**:

No warning triggered.

**Actual Behavior**:
```
Use nameof in place of string literal 'Xxx'
```

**Comment by kentcb** [comment]
@ivanbasov sorry, I think I created this against the wrong repository. Would you mind moving it to dotnet/roslyn-analyzers as appropriate?

**Comment by ivanbasov** [comment]
@kentcb, thank you! Please move it to dotnet/roslyn-analyzers

**Comment by kentcb** [comment]
Moved to https://github.com/dotnet/roslyn-analyzers/issues/1452.
